mlir/lib/Dialect/Index/IR/IndexOps.cpp

```diff
@@ -594,6 +594,8 @@ static bool compareSameArgs(IndexCmpPredicate pred) {
   case IndexCmpPredicate::ULT:
     return false;
   }
+  assert("unknown predicate in compareSameArgs" == nullptr);
```

Comment:
llvm_unreachable, and delete dead return

Reply:
Yes, see above. This is already implemented and pushed, but for some reason github is still processing the update...